packages/plugin-core/src/utils/ConfigMigration.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  CURRENT_CONFIG_VERSION,
+  IntermediateDendronConfig,
+  MigrationEvents,
+} from "@dendronhq/common-all";
+import {
+  MigrationChangeSetStatus,
+  WorkspaceService,
+} from "@dendronhq/engine-server";
+import _ from "lodash";
+import * as vscode from "vscode";
+import { AnalyticsUtils } from "./analytics";
+
+/**
+ * Utility methods that are only used in plugin-core.
+ * Bulk of the actual migration related logic is in [[../packages/engine-server/src/migrations/utils.ts]]
+ */
+export class ConfigMigrationUtils {
+  static maybePromptConfigMigration(opts: {
+    dendronConfig: IntermediateDendronConfig;
+    wsService: WorkspaceService;
+    currentVersion: string;
+  }) {
+    const { dendronConfig } = opts;
+    if (dendronConfig.version !== CURRENT_CONFIG_VERSION) {
+      ConfigMigrationUtils.showConfigMigrationConfirmationMessage(opts);
+      return true;
+    }
+    return false;
+  }
+
+  static showConfigMigrationConfirmationMessage(opts: {
+    dendronConfig: IntermediateDendronConfig;
+    wsService: WorkspaceService;
+    currentVersion: string;
+  }) {
+    vscode.window
+      .showInformationMessage(
+        "We have detected a legacy configuration in dendron.yml. Would you like to run a migration?",
```

Comment:
reword -> "We notice that your dendron.yml is not up to date. Would you like to migrate to the latest configuration version?" (A backup of your current configuration will be made before the migration)